### PR TITLE
Android: Keep reference to VideoCapturers and dispose of them with media stream

### PR DESCRIPTION
**Pull request by stwiname**
Fixes https://github.com/oney/react-native-webrtc/issues/158

And might also fix (I think this might be device specific): https://github.com/oney/react-native-webrtc/issues/116 https://github.com/oney/react-native-webrtc/issues/101

### CODE REVIEW

File: android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java [hunk 1]
```diff
@@ -327,11 +330,14 @@ public void getUserMedia(ReadableMap constraints,
...
                     videoSource = mFactory.createVideoSource(
                             videoCapturer, videoConstraints);
+
+                    trackId = getNextTrackUUID();
+
+                    mVideoCapturers.put(trackId, videoCapturer);
                 }
             }
 
             if (videoSource != null) {
```
Comment (zxcpoiu) [inline android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java]: never mind, I've made a PR to your branch

**Comment by zxcpoiu** [comment]
I have tested it on `RCTWebRTCDemo`.

* compilation passed
* switched back and force passed

but I don't have any further video cases.

**Comment by zxcpoiu** [comment]
@maximovicd

can you test it on this?

**Comment by zxcpoiu** [comment]
another question:

does anyone know:

1. should we do `removeVedioCapturer` in `peerConnectionClose` and `peerConnectionRemoveStream` also?

2. would `String onRemoveStream()` be called automatically after `peerConnectionRemoveStream`?

**Comment by stwiname** [comment]
I'm not sure these are valid use cases but i think:

1. Users might want to keep displaying a local stream after a peer connection has been removed
1. Users might attach a local stream to multiple peer connections

So cleaning up a local stream at the end of a peer connection might not be desired behavior

**Comment by zxcpoiu** [comment]
@stwiname you are right, makes scene. thanks your reply and PR.

I think this PR is ready for merge.

ping @lyubomir 

about comments by @maximovicd from #158 
>  it might be possible to create multiple tracks from one capturer, in which case when all tracks are dropped, we should drop the main capturer.

this doesn't seem to be happen based on current source code. ( we always create `VideoCapturer` before create video track )

but it this behavior changes in the future, we should have this case in mind and might be change to counter-based refrences

**Comment by maximovicd** [comment]
Great job everyone, this seems to fix it for me.

@zxcpoiu Yes, you are right, but I am not sure if you should be able to create another `VideoCapturer` on the same camera. I don't think you will be allowed to do that because one capturer takes hold of one camera, in that case, we should maybe think of just creating one more track from existing capturer (if that's possible too) and do reference counting. This is just me guessing things, but I can try that out when I have more time and let you know.

**Comment by zxcpoiu** [comment]
thanks @maximovicd 

OK, let's get this merged.

**Pull request merged**